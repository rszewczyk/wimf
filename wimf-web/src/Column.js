// @flow
import React from "react";
import { css } from "glamor";

type Width = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8;

type ColumnProps = {
  children: any,
  width: Width
};

const columnStyle = css({
  display: "flex",
  flexDirection: "column"
});

function getWidth(w: Width) {
  return `${12.45 * w}%`;
}

export default function Column(props: ColumnProps) {
  return (
    <div
      {...columnStyle}
      style={{ width: props.width ? getWidth(props.width) : getWidth(2) }}
    >
      {props.children}
    </div>
  );
}
