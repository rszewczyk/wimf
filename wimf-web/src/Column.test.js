import React from "react";
import renderer from "react-test-renderer";
import Column from "./Column";

it("renders correctly", () => {
  const widthOne = renderer
    .create(<Column width={1}><p>Hello</p></Column>)
    .toJSON();
  expect(widthOne).toMatchSnapshot();

  const widthThree = renderer
    .create(<Column width={3}><div>Foobar!!</div></Column>)
    .toJSON();
  expect(widthThree).toMatchSnapshot();
});
