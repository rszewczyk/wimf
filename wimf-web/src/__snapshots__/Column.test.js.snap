// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  data-css-1t62idy=""
  style={
    Object {
      "width": "12.45%",
    }
  }
>
  <p>
    Hello
  </p>
</div>
`;

exports[`renders correctly 2`] = `
<div
  data-css-1t62idy=""
  style={
    Object {
      "width": "37.349999999999994%",
    }
  }
>
  <div>
    Foobar!!
  </div>
</div>
`;
